pkg/database/authorized_app_stats.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package database
+
+import (
+	"time"
+
+	"github.com/jinzhu/gorm"
+)
+
+// AuthorizedAppStats represents statistics related to a user in the database.
+type AuthorizedAppStats struct {
+	gorm.Model
+	Date            time.Time `gorm:"unique_index:idx_date_app_realm"`
+	AuthorizedAppID int64     `gorm:"unique_index:idx_date_app_realm"`
+	AuthorizedApp   AuthorizedApp
```

Comment:
only necessary if we're going to follow the relation from this record

Reply:
without both fields, no fields were being placed as columns by gorm. Only appid is stored in gorm. But the model has authorizedapp. I can just track id. is that what you mean?

Reply:
removed object relations from app stats and user stats.